src/Hl7.Fhir.Core/Rest/TypedEntryResponseToBundle.cs

```diff
@@ -0,0 +1,134 @@
+                        {
+                            VersionId = ri.VersionId
+                        };
+                        result.Resource.ResourceBase = ri.BaseUri;
+                    }
+                }
+                else
+                {
+                    if(entry.TypedElement != null)
+                    {
+                        try
+                        {
+                            result.Resource = new BaseFhirParser(parserSettings).Parse<Resource>(entry.TypedElement);
+
+                            if (result.Response.Location != null)
+                                result.Resource.ResourceBase = new ResourceIdentity(result.Response.Location).BaseUri;
+                        }
+                        catch (AggregateException ae)
+                        {
+                            throw ae.GetBaseException();
+                        }
+                    }
+                }
+            }
+            return result;
+        }
+
+        internal static bool IsBinaryResponse(string responseUri, string contentType)
+        {
+            if (!string.IsNullOrEmpty(contentType)
+                && (ContentType.XML_CONTENT_HEADERS.Contains(contentType.ToLower())
+                    || ContentType.JSON_CONTENT_HEADERS.Contains(contentType.ToLower())
+                )
+                )
+                return false;
+
+            if (ResourceIdentity.IsRestResourceIdentity(responseUri))
+            {
+                var id = new ResourceIdentity(responseUri);
+
+                if (id.ResourceType != ResourceType.Binary.ToString()) return false;
+
+                if (id.Id != null && Id.IsValidValue(id.Id)) return true;
+                if (id.VersionId != null && Id.IsValidValue(id.VersionId)) return true;
+            }
+
+            return false;
+        }
+
+        internal static Binary MakeBinaryResource(byte[] data, string contentType) =>
+            new Binary
+            {
+                //Content is for STU3, from R4 Content has changed into Data
+                Content = data,
+                ContentType = contentType
+            };
+
+        private class Body
+        {
+            public byte[] Data;
+        }
+
+
+        public static byte[] GetBody(this Bundle.ResponseComponent interaction) => interaction.Annotation<Body>()?.Data;
+
+        public static string GetBodyAsText(this Bundle.ResponseComponent interaction)
+        {
+            var body = interaction.GetBody();
+            return body != null ? HttpUtil.DecodeBody(body, Encoding.UTF8) : null;
+        }
+        
+        internal static void SetBody(this Bundle.ResponseComponent interaction, byte[] data)
+        {
+            interaction.RemoveAnnotations<Body>();
+            interaction.AddAnnotation(new Body { Data = data });
+        }
+
+        internal static void SetHeaders(this Bundle.ResponseComponent interaction, Dictionary<string, string> headers)
```

Comment:
I like to have the signature of the method with interfaces, so the client can use it also with his own classes.
```suggestion
        internal static void SetHeaders(this Bundle.ResponseComponent interaction, IDictionary<string, string> headers)
```

Reply:
Done